tests/test_reportportal_config.py

```
@@ -0,0 +1,98 @@
+            REPORTPORTAL_PROJECT="my-project",
+            ENABLE_REPORTPORTAL="false",
+        )
+        with patch.dict(os.environ, env, clear=True):
+            settings = Settings(_env_file=None)
+            assert not settings.reportportal_enabled
+
+    def test_rp_warns_when_explicitly_enabled_but_url_missing(self, caplog):
+        """When enable_reportportal=True but url is missing, warn."""
+        env = _build_env(
+            ENABLE_REPORTPORTAL="true",
+            REPORTPORTAL_API_TOKEN="rp-token",  # pragma: allowlist secret
```

Ruff (0.15.9)

[error] Possible hardcoded password assigned to argument: "REPORTPORTAL_API_TOKEN"

(S106)